### ISSUE TITLE
It is not easy to access content and core contribution guides on the website.

### ISSUE THREAD

**Issue by GypsyDangerous** [CodingTrain/website#1750]
It is very easy to access the community contribution guide because it is on every coding challenge's page, but the other guides like the core contribution guide, and content contribution guide are not easy to access from website. I think it would be good to add some way to access all the guides.

**Comment by GypsyDangerous** [comment]
I think when the wiki was moved to markdown files those files were never given a direct link from the website.

**Comment by GypsyDangerous** [comment]
@vimkat @phucduong86

**Comment by phucduong86** [comment]
@GypsyDangerous which wiki are you refer to?
I'm not really involved with the website content arrangement other than accessibility issues but it seems that the links to core/content contribution guides are both available on the README page of this github repo.

**Comment by GypsyDangerous** [comment]
I'm talking about the [content contribution guide](https://thecodingtrain.com/Guides/content-contribution-guide.html) which is available form the website readme and the [core contribution guide](https://thecodingtrain.com/Guides/core-contribution-guide.html) which is not. While the content contribution guide is available from the readme I think it would be good to have links on the actual website

**Comment by phucduong86** [comment]
Actually looks like all the guides are available in this same page on the website:
https://thecodingtrain.com/Guides/

I do agre they are not intuitive to find from the Navigation Menu, and since the navigation menu is getting crowded it might be good to add the Guide links as another section on the home page.
@shiffman @vimkat ?

**Comment by Fraasi** [comment]
Here we go again 😁 #367
How about having a dropdown or two in the nav menu so we could group some of the links?

**Comment by GypsyDangerous** [comment]
thats a good idea, we could group coding challenges and coding in the cabana into one drop down and group tutorials and courses into another.

**Comment by violetcraze** [comment]
Been thinking about navigation a lot recently due to adding some new pages. The way I see it there's a couple goals of the navigation on the site.

1. It's beginner friendly. @shiffman has talked about this in the past (#357) but I think it's important to direct new people to videos that will match their skill level. The current site differentiates this by splitting up tutorials and courses.

2. It's easy to find videos, code, and other content on the site. Our current system leaves out Guides and GuestTutorials from the navigation. And there's some code missing for older courses (probably a separate issue)

3. Accessible to screen readers. As discussed in #1759 there's more we could do to make sure it's easy to navigate the site with a keyboard or screen reader.

I think the number of new and hidden pages kind of necessitates drop downs in the navigation bar. I'd be happy to work on making a drop down navigation method accessible. But would like input from other's on structuring the navigation and how best to meet goal 1 and 2.

Here's my current idea:
```
- Home
- Challenges
  - Coding Challenges
  - Coding in the Cabana
- Learning
  - Learn to Code with p5.js
  - Get Machines Learning in ml5.js
  - Learn to Code with Processing
- About
  - About the Coding Train
  - How to Contribute
  - FAQ
  - More Guides
- Courses
  - Nature of Code
  - Data and APIs
  - ect.
- More
  - Guests Tutorials
  - other stuff I've forgotten
```

**Comment by shiffman** [comment]
I really like this! Thank you @violetcraze for putting these thoughts together, I think the priorities you list are dead on. I might add "ability for viewers to submit their own versions inspired by the videos" as a priority but I think we have a "good enough" system for that right now.

The only change I would consider to the above is the distinction between "Learning" and "Courses", it doesn't feel particularly meaningful right now. As is I would prefer just one "Learning" category. Also, while I do teach courses at NYU the YouTube playlists aren't really classes you can enroll in. The closest thing we have to that is membership. So I might just go with Learning and have everything under that. If we want to distingusih I might do:

```
- Beginners
  - Learn to Code with p5.js
  - Learn to Code with Processing
  - Git and GitHub for Poets
  - Get Machines Learning in ml5.js (maybe not beginners?)

- Learning
  - everything else
```

But maybe everything under learning with some sort of "emoji" tagging system for what is good for total beginners is best?

**Comment by vimkat** [comment]
Yes, I totally agree with @violetcraze and @shiffman!

There's already the idea to give - at least each coding challenge - a skill level rating so people can filter based on that. The "guides" have just been migrated to the repo itself and I thought it would be good to integrate them into the website/design. That's why you can't really "find" them apart from linking to them directly. Maybe making them a core feature and expanding the content (how to get Processing/P5 set up, etc.) might be worth a try?

At some point it might also be a good idea to separate the website from the actual code (from the videos) to keep everything organized, searchable and maintainable?

**Comment by violetcraze** [comment]
Thanks for the feedback. @shiffman I like the idea of beginners and learning as titles. Think you might be right that machine learning would fall more under advanced topics as it would be much easier to follow if someone has JavaScript experience.

@vimkat love the idea of a skill level for challenges but that might be better broken out into it's own issue.

Also in terms of splitting code and the website it sounds like that would be a bigger change definitely something we could shoot if project trainsite ever becomes the main site. But if we were to work on it before then, I think the approach I would suggest is to make a separate code repository with all the code and get all the links pointing to that one before we start removing code from the site repository. Would have the added overhead of making and reviewing more pull requests but might make it easier to set up some formatting and header standards that are automated for all code examples.

I think over the next week I'll work on making a drop down just for the current courses section and then once it's working and accessible we could move on to restructuring.

**Comment by vimkat** [comment]
@violetcraze Sounds like a great plan! Thank you very much! 🎉 

Yes, skill levels would be a more complex feature to implement and should be done separately. I'd even go as far as saying that this should be a feature that gets added once Project Trainsite is finally taking off and being able to offer more complex filtering/sorting/etc. is one of the main drivers to switch to a database-driven system. Trying to put it into the current Jekyll setup would just be a nightmare...
